src/keylessBackup/KeylessBackupProgress.tsx

```diff
@@ -24,21 +29,77 @@ const TAG = 'keylessBackup/KeylessBackupProgress'
 function KeylessBackupProgress({
   route,
 }: NativeStackScreenProps<StackParamList, Screens.KeylessBackupProgress>) {
-  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
-
-  const { t } = useTranslation()
-
   // Disable back button on Android
   useEffect(() => {
     const backPressListener = () => true
     BackHandler.addEventListener('hardwareBackPress', backPressListener)
     return () => BackHandler.removeEventListener('hardwareBackPress', backPressListener)
   }, [])
 
-  // TODO(ACT-781): Implement Restore flow designs
   if (route.params.keylessBackupFlow === KeylessBackupFlow.Restore) {
-    return <></>
+    return <Restore />
+  } else {
+    return <Setup />
   }
+}
+
+function Restore() {
+  const keylessBackupStatus = useSelector(keylessBackupStatusSelector)
+  const localCurrencyCode = useLocalCurrencyCode()
+
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  switch (keylessBackupStatus) {
+    case KeylessBackupStatus.InProgress: {
+      return InProgress(t('keylessBackupStatus.restore.inProgress.title'))
+    }
+    case KeylessBackupStatus.RestoreZeroBalance: {
+      return (
+        <SafeAreaView>
+          <Dialog
+            title={
+              <Trans i18nKey="importExistingKey.emptyWalletDialog.title">
+                <TokenDisplay
+                  localAmount={{
+                    value: new BigNumber(0),
+                    currencyCode: localCurrencyCode,
+                    exchangeRate: '1',
+                  }}
+                  showLocalAmount={true}
+                  amount={new BigNumber(0)}
+                />
+              </Trans>
+            }
+            isVisible={true}
+            actionText={t('importExistingKey.emptyWalletDialog.action')}
+            actionPress={() => {
+              ValoraAnalytics.track(KeylessBackupEvents.cab_restore_zero_balance_accept)
+              dispatch(keylessBackupAcceptZeroBalance())
```

Comment:
This doesn't navigate away from the screen immediately right? We've had issues in the past where the screen hangs because the modal isn't dismissed completely before navigating

Reply:
no it doesn't, there are a few async methods that complete before navigating. I will check this though.